File: Kymdan_front_end/src/app/service/product.service.ts
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {HttpClient} from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class ProductService {
  public readonly API: string = 'http://localhost:8080/product';

  constructor(
    public http: HttpClient
  ) {
  }

  findProductByTypeID(typeID): Observable<any> {
    return this.http.get(this.API + '/view/' + typeID);
  }

  saveCart(userName, productID, productInformation): Observable<any> {
    return this.http.get(this.API + '/save-cart/' + userName + '/' + productID + '/' + productInformation);
  }
}

File: Kymdan/src/main/java/com/kymdan/backend/services/product/ProductService.java
package com.kymdan.backend.services.product;

import com.kymdan.backend.entity.Product;
import com.kymdan.backend.model.MessageDTO;

import java.util.List;

public interface ProductService {
    List<Product> findProductByType(Long typeID);

    MessageDTO saveCart(String userName, Long productID, String productInformation);
}
